### ISSUE TITLE
Make the PAM50 in the same coordinate system as the MNI template

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#2085]
As shown in the Figure 4 of De Leneer et al., the PAM50 template should be in the same coordinate system as the MNI template. 

@benjamindeleener what does it take to do so? Only update the q/sform or is there anything more fancy to do?

Also see: https://github.com/neuropoly/template/issues/6#issuecomment-442465045

**Comment by benjamindeleener** [comment]
@jcohenadad Merging the two templates could work with a rigid transformation if the objective is just to put the PAM50 in the MNI space. Overall, it is more complicated if we want a final product as the registration of the two templates over the brainstem was not perfect on the latest iteration of the PAM50. I am working on it.

**Comment by jcohenadad** [comment]
@benjamindeleener yes, as a first step, the goal is to simply get the s/qform right so that people can use the same coordinate system between the MNI and the PAM50. Several users have contacted me with this request so we should implement it. It would ideally be ready for v3.3 (next release, in ~1w) because it will break compatibility with 3.2.7. Let me know if the timeline is good on your side.

**Comment by benjamindeleener** [comment]
Timeline is good for me.

**Comment by benjamindeleener** [comment]
The new template is ready, that is in the same space as the ICBM152. Note that the two templates are not registered yet.

Basically, I used a simple translation on each file from the template (only changing the header) using the location (in physical space) of a common anatomical landmark. To calculate the translation, I used the tip of the central canal, as shown in the figures below.

Anatomical landmark in the ICBM152 space (red dot)
![image](https://user-images.githubusercontent.com/7757117/49654198-2fe33500-fa05-11e8-8a86-5c7b95d80461.png)

Anatomical landmark in the PAM50 space (red dot)
![image](https://user-images.githubusercontent.com/7757117/49654244-4f7a5d80-fa05-11e8-8f17-74f2e216c344.png)

The result of the translation
![image](https://user-images.githubusercontent.com/7757117/49654345-8f414500-fa05-11e8-88b4-154711dd2df3.png)

**Comment by jcohenadad** [comment]
Thank you @benjamindeleener!!! 

So, my understanding is that in the [NIMG paper](https://www.ncbi.nlm.nih.gov/pubmed/29061527), the PAM50 looks better registered to the MNI template at the brainstem level because it was a more recent version of the PAM50, different from the version which is currently implemented in SCT. If we want to use this more recent version, then we would need to re-create the segmentations and atlases associated with this new version.

A few questions/suggestions:
- How does the new PAM50 version compare to the existing PAM50? What is different in the methods? What would be the pros of the new version (apart from having the brainstem closer to the MNI template)?
- How about we crop the PAM50 where the MNI template starts? I see two pros: (i) smaller file (faster processing, quicker to download, etc.) and (ii) no more issue with brainstem registration, because the MNI template already has the brainstem. A cons is that some researchers in the past asked us to include the brainstem, but this cons is mitigated now that the PAM50 and MNI templates are in the same coordinate system. See below an animation showing where the PAM50 could be cropped:

![pam50_mni](https://user-images.githubusercontent.com/2482071/49658478-2612ff00-fa10-11e8-9fc7-88b99803cef1.gif)

Suggestions from everyone are welcome!

**Comment by jcohenadad** [comment]
Comment from Falk Eippert:
> when looking at your animation, wouldn’t bringing the pam50 into mni space lead to ‘anatomical inconsistency’, because if you were indeed to cut the pam50 at the top of the cervical cord, there would be a kind of kink from brainstem to spine it seems, as one of the two templates would need to be (as least) rotated to match the other? i absolutely love the idea of being able to report mni coordinates, but if they refer to something that is anatomically implausible, is that desirable? 

This is a tricky point. By making the PAM50 "straight", we already accepted to reduce anatomical plausibility, for the benefits of processing and visualization. That being said, now remains the question of "connecting" the cord and the brainstem. In the current PAM50, the brainstem is not anatomically consistent either because it was part of the straightening. So, the animation showed above would in fact bring more anatomical consistency. However, I agree that we could rotate the cord a bit such that the angle between the bottom part of the brainstem (medulla oblongata) and the top part of the cropped PAM50 (top C1) is close to zero. It would make the visualisation less friendly though (i.e., when looking at a coronal view of the PAM50 in the MNI space, we would not have a full view of the cord).

**Comment by fprados** [comment]
I agree that we might need to rotate the cord, however the whole image will look strange from a coronal view. Although, if the cropping point is the top of C1, the only issue is from a visualisation perspective, isn't it? From a processing point of view, you will process brain and cord stuff separate. Then, we can do a tool that just for visualisation proposes that rotates the brain or the coord, depending what you need.

**Comment by jcohenadad** [comment]
> Although, if the cropping point is the top of C1, the only issue is from a visualisation perspective, isn't it? From a processing point of view, you will process brain and cord stuff separate. Then, we can do a tool that just for visualisation proposes that rotates the brain or the coord, depending what you need.

Thank you for your input @fprados. Indeed, we are mostly talking about a visual perspective issue here. However, I would like to stress that the brainstem currently represented in the PAM50 template (in the animation above) is **not** consistently positioned with the upper cervical part, and has already been manipulated to be straightened. So, even though we are tempted to "rotate" the PAM50 a bit to align the PAM50_brainstem to the MNI_brainstem, that would still produce a "wrong" brainstem-cord continuity. Also as you said, ultimately there are already tools that can process the brain/brainstem and the cord separately. Then, the MNI-PAM50 merged space would be used solely for the purpose of final visualization and interpretation. Thus, if visualization is important here, I would rather have a cord that is aligned in the coronal plane.

**Comment by benjamindeleener** [comment]
I agree with the fact that we always process brain, brainstem and spinal cord separately and for visualization purposes and reporting coordinates in the MNI space, I think it is easier to work with a straight template without tilting the spinal cord.

If we aim at providing a template that is anatomically plausible, we could even change the template so that it integrates the average curvature of the population. However, this would increase interpolation errors related to curvature when analysis data at a group level, so I would not suggest it.

As we perform analysis in brain and spinal cord in a separate fashion, I would suggest focussing on usability and visualization, and so to provide a spinal cord template that is straight when looking at brain + spinal cord images. By deforming the region between the top of the spinal cord and the brainstem, we are making the study of this region more difficult as anatomically non-plausible and deformed. However,  I think the other reasons for straightening the template are more important.

**Comment by jcohenadad** [comment]
Thanks for your input @benjamindeleener. So it seems like the consensus so far is **not** to rotate the cord. And how about cropping above C1 (as shown in my anim)? Do you agree with that?

**Comment by benjamindeleener** [comment]
I would suggest having at least a little bit of overlap between the ICBM152 and the PAM50, as the image quality of the spinal cord in the ICBM152 template is not great and it's contours are not well defined.

**Comment by benjamindeleener** [comment]
2-3 slices above a common anatomical landmark would be enough.

**Comment by benjamindeleener** [comment]
Sorry I was confused. Yes, 2-3 slices after registration will be enough. I'll update the template tomorrow.

**Comment by benjamindeleener** [comment]
I updated the template with the required changes. Below is the results of the alignment. Note that I cropped the template above the C1 label (top of C1) so that registration to the template is performed appropriately.

![screenshot from 2018-12-11 09-30-55](https://user-images.githubusercontent.com/7757117/49807207-7dbeac80-fd27-11e8-9af2-3207e3ad5aaa.png)

I also fixed a few things:
1. the file info_label.txt in the folder template/ did not contain all the file. I updated it.
2. the labels in PAM50_label_discPosterior.nii.gz were shifted by 1 as shown in the figure below

![wrong_label_posterior](https://user-images.githubusercontent.com/7757117/49807160-67185580-fd27-11e8-93d7-71d536ddbed2.png)

Are we good with this result?
